### PR TITLE
batch Map

### PR DESCRIPTION
**Pull request by comzyh**

### CODE REVIEW

File: records/chain_reader.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+// Pop Pop one record from ChainReader
+func (cr *ChainReader) Pop() *Record {
+	return cr.readers[cr.current].Pop()
+}
+
+// HasNext HasNext
+func (cr *ChainReader) HasNext() bool {
+	if cr.current >= len(cr.readers) {
```
Comment (Lcch) [inline records/chain_reader.go]: it's redudent

File: records/chain_reader.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package records
+
+// ChainReader Chain some reader to one
+type ChainReader struct {
+	RecordReader
+	readers []RecordReader
+	current int
+	closed  []bool
```
Comment (Lcch) [inline records/chain_reader.go]: Why does it need closed indicator?

File: records/chain_reader.go (hunk 1 above)
Comment (comzyh) [inline records/chain_reader.go]: yes

File: records/chain_reader.go (hunk 2 above)
Comment (comzyh) [inline records/chain_reader.go]: every reader is closed after `HaxNext`return false, we should not close reader again after the reader is already closed.

**Pull request merged**